group_vars/all.yml

```diff
@@ -0,0 +1,58 @@
+cluster_name: "Yugabyte_cluster"
+type_instance: "t2.micro"
+region: "eu-central-1"
+cloud: aws
+# aws.networking
+aws_region:     'eu-central-1'
+vpc_name:       'Yugabyte_VPC_Res'
+vpc_cidr_2:     '185'
+vpc_cidr: '10.{{vpc_cidr_2}}.0.0/16'
+
+subnets:
+  - type: public
+    cidr: "10.{{vpc_cidr_2}}.10.0/24"
+    az: a
+  - type: public
+    cidr: "10.{{vpc_cidr_2}}.11.0/24"
+    az: b
+  - type: public
+    cidr: "10.{{vpc_cidr_2}}.21.0/24"
+    az: c
+
+route_name: "yugabyte_route"
+instances_per_subnet: 2
+
+# Internet Gateway Name
+igw_name: "Yugabyte_igw"
+ 
+securitygroup_name: "Yugabyte_security_group"
+
+#master_dir: "~/master"
+#tserver_dir: "~/tserver"
+
+ec2_tag: "Yugabyte_nodes"
+ansible_user: centos 
+#The local path to which we would save our EC2 Private Key
+ec2_key_directory: "./"
+keypair_name: "ec2_key_pair"
+key_location: "~/Documents/yugabyte_exam/key.ppk"
```

Comment:
yes of course, key_location_variable is the customizable path where the key is going to be stored once it has been generated automatically.

Reply:
I have made a small improvement here:
If the keypair name already exists in amazon, then it would take ec2_key_directory variable as the key location, otherwise it will create it and store it within the specified ec2_key_directory variable.